polygon/heimdall/entity_fetcher.go

```diff
@@ -116,11 +117,9 @@ func (f *entityFetcherImpl[TEntity]) FetchAllEntities(ctx context.Context) ([]TE
 		}
 
 		for _, entity := range entitiesPage {
-			if entity.RawId() == 0 {
-				// TODO use workaround from heimdall.Heimdall FetchCheckpointsFromBlock
-				//      to set entity id or fix heimdall API to return "id" as part of json
-				panic("unexpected 0 id for entity - pending fix")
-			}
+			entity.SetRawId(checkpointId)
```

Comment:
the results are not ordered by id unfortunately. please add a new loop after the sort call below (also please move the checkpointId variable there)